rage/src/bin/rage-keygen/cli.rs

```diff
@@ -35,6 +36,7 @@ pub(crate) struct AgeOptions {
     #[arg(short, long)]
     #[arg(value_name = fl!("output"))]
     #[arg(help = fl!("keygen-help-flag-output"))]
+    #[arg(value_hint = ValueHint::AnyPath)]
```

Comment:
`ValueHint::AnyPath` is documented as "Any existing path". Is that going to have the correct behaviour here, given that the output file will in most cases not exist yet? I guess the idea is that it will let you auto-complete a directory, after which you can type whatever filename you want.

Note however that `rage-keygen` does not allow overwriting existing files (because we never want to risk overwriting existing key material), so providing a file path here is always incorrect. So this should be:
```suggestion
    #[arg(value_hint = ValueHint::DirPath)]
```

Also, do these autocompletes interfere at all with the user specifying stdout via `-`?

Reply:
> Note however that `rage-keygen` does not allow overwriting existing files (because we never want to risk overwriting existing key material), so providing a file path here is always incorrect.

You're right. I will change it to `DirPath`.
> Also, do these autocompletes interfere at all with the user specifying stdout via `-`?

What I'm getting on zsh after typing `rage-keygen -o -` and then TAB is `-- no matches found --`. So it is looking for a directory (I changed it to `ValueHint::DirPath`) in the current directory that starts with `-`. However if I press space then it accepts that my output file is `-` and the other completions work normally after that.
I would say this is exactly the expected behaviour, or I don't really get your question.